src/components/ImagePreviewer.js

```diff
@@ -0,0 +1,176 @@
+import { stringify } from "querystring";
+import { decode } from "base58";
+
+const noop = () => {};
+
+export const of = src => Promise.resolve({ src });
+
+export const makeSrcResolver = ({ src }) =>
```

Comment:
Maybe just use the verb `resolve`, it returns a `Promise` that does the resolving.

Reply:
Renamed to `resolveSrcToImageUrl`